From 3151a1c6e7deb034063e0b7ba951abc8bc5d9417 Mon Sep 17 00:00:00 2001
From: lhdjply <lhdjply@126.com>
Date: Wed, 30 Apr 2025 16:44:39 +0800
Subject: [PATCH] fix riscv64 build error

Signed-off-by: lhdjply <lhdjply@126.com>
---
 include/my_cpu.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/include/my_cpu.h b/include/my_cpu.h
index bd5fca5f..c49379c7 100644
--- a/include/my_cpu.h
+++ b/include/my_cpu.h
@@ -96,11 +96,13 @@ static inline void MY_RELAX_CPU(void)
 #elif defined(_ARCH_PWR8)
   /* Changed from __ppc_get_timebase for musl and clang compatibility */
   __builtin_ppc_get_timebase();
-#elif defined __GNUC__ && defined __riscv
-  __builtin_riscv_pause();
-#elif defined __GNUC__
+#elif defined __GNUC__ && (defined __arm__ || defined __aarch64__)
   /* Mainly, prevent the compiler from optimizing away delay loops */
   __asm__ __volatile__ ("":::"memory");
+#else
+  int32 var, oldval = 0;
+  my_atomic_cas32_strong_explicit(&var, &oldval, 1, MY_MEMORY_ORDER_RELAXED,
+                                  MY_MEMORY_ORDER_RELAXED);
 #endif
 }
 
-- 
2.47.2

